### PR TITLE
Fixes #34305 - stop creating settings in DB

### PR DESCRIPTION
**Pull request by ezr-ondrej**
With the setting DSL, we hold all the necessary values for settings in the memory.
We can now stop creating records for the settings in the database.

Needs:
- [x] tasks fix: theforeman/foreman-tasks/pull/682

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
Draft for now, I believe it does not affect many plugins now (I didn't find any that doesn't use `default_settings`), but it is still a major change so I'd wait for 3.3

**Comment by ekohl** [comment]
For what it's worth, I think this is something we shouldn't do. I like having everything in the DB: it ensures the view is consistent across multiple application servers, even if there are subtle differences in application code or where it's derived from hostnames.

@laugmanuel @timogoebel perhaps you could also weigh in since AFAIK you run that scenario in production.

**Comment by ezr-ondrej** [comment]
> I like having everything in the DB

well, define everything, because if default values, then the ship has sailed, because the defaults are served from memory already anyway and just still hang around in database as well :)

IF there is discrepancy between two servers we should fix such instances.
Tho you're correct it was consistent accross aplication servers prior that, it was less consistent across restarts of the servers, so say you've restarted the secondary application server and while your app is still running, your default value for some setting have changed, because the secondary server has overidden the primary server's value with its value. Ugh... I do not think that "it was ok to have different values accross application servers, because database". And this approach has many advantages, and just doesn't solve the problem that has already been there (that should have been fixed for sure to have a stable defaults accross application servers). And I'm all for addressing that before this :) I'd just not block that completely :))

**Comment by laugmanuel** [comment]
> @laugmanuel @timogoebel perhaps you could also weigh in since AFAIK you run that scenario in production.

Yeah, we do run multiple application servers but I would need to check if this change would affect us.

Just to confirm: this will only be a problem if we use the default value and the default value would be different between servers?

**Comment by ezr-ondrej** [comment]
> Just to confirm: this will only be a problem if we use the default value and the default value would be different between servers?

Yes exactly :)

**Comment by adamruzicka** [comment]
Needs a rebase. What remains to be done here to un-draft it?

**Comment by ezr-ondrej** [comment]
I'd like to get https://github.com/theforeman/foreman/pull/9150 in first as that will cleanup the initialization of old settings. Then I believe this can be rebased and undrafted. Tho to fix weird issues with Rails initialization, we will need https://github.com/theforeman/foreman/pull/9155

File: app/models/setting.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'resolv'
 
 class Setting < ApplicationRecord
-  audited :except => [:name, :description, :category, :settings_type, :full_name, :encrypted], :on => [:update]
+  audited :except => [:name, :category], :on => [:update]
```
Comment (ezr-ondrej) [inline app/models/setting.rb]: **Food for thought:** we are not yet ready to drop category as some settings still may rely on the column existence to validate values (looking at you Katello!), but is it worth delaying this to Foreman 3.4, when we are saying setting subclassing is deprecated, or will we drop that separately?

**Comment by ofedoren** [comment]
@ezr-ondrej, test failures seem to be related...

**Comment by ofedoren** [comment]
@ezr-ondrej, still failing :/

**Comment by ezr-ondrej** [comment]
> @ezr-ondrej, still failing :/

I think foreman tests should be fixed now, but katello failures are worrisome. I thought this was fixed and it doesn't look like it :/

**Comment by ofedoren** [comment]
[test katello]

**Review by ShimShtein — DISMISSED** [review]

**Comment by ShimShtein** [comment]
@ezr-ondrej Last rebase please?

**Comment by ezr-ondrej** [comment]
@ShimShtein happy to, just Katello still struggle with the require order issue, should I notify Katello team prior merge?

**Comment by ShimShtein** [comment]
@ezr-ondrej please do. I can wait a bit longer with this PR :smile:

**Comment by jeremylenz** [comment]
It was foreman-tasks.. :smiley:   https://github.com/theforeman/foreman-tasks/pull/682

**Comment by adamruzicka** [comment]
foreman-tasks-6.0.2 was just released to rubygems [test katello]

**Comment by ezr-ondrej** [comment]
We are IMHO ready here (katello is :green_apple:) should we merge so it's in early in the devel process? @ShimShtein ?

**Review by jeremylenz — APPROVED** [review]
:+1: from Katello-land!

**Comment by ShimShtein** [comment]
Merged, thanks @ezr-ondrej !

**Pull request merged**